Support opening trace files through a dialog

ngondalia opened the pull request:
This commit adds another button for opening trace files through a dialog. Futhermore, the theia-trace-explorer-placeholder-widget can be configured to render either one or both buttons (Open Trace folder, Open Trace File).

Fixes #1053

Signed-off-by: Neel Gondalia <ngondalia@blackberry.com>

ngondalia commented:
https://github.com/eclipse-cdt-cloud/theia-trace-extension/assets/113199503/86ab53bb-d24c-4745-a756-46372c044d71

hriday-panchasara commented on theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer-toolbar-commands.ts:
```diff
@@ -44,10 +44,16 @@ export namespace TraceViewerToolbarCommands {
     };
     export const OPEN_TRACE: Command = {
```
Maybe rename OPEN_TRACE command and id to OPEN_TRACE_FOLDER. I always seem to get confused with all the Open Trace Command references in the codebase

ngondalia replied:
Yeah it happens to me as well - i will rename them

hriday-panchasara commented on theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer-toolbar-contribution.tsx:
```diff
@@ -330,6 +343,12 @@ export class TraceViewerToolbarContribution implements TabBarToolbarContribution
+            id: TraceViewerToolbarCommands.OPEN_TRACE_FILE.id,
+            command: TraceViewerToolbarCommands.OPEN_TRACE_FILE.id,
+            tooltip: TraceViewerToolbarCommands.OPEN_TRACE_FILE.label,
+            priority: 7
```
The item registered with the command 'TraceViewerToolbarCommands.MARKER_SETS.id' also has the same id (7). Although these items won't cause conflict as they are not displayed in the same toolbar, you might consider adjusting the priorities so that its unique for each item

ngondalia replied:
Ah yes - i missed that! I will make them unique

hriday-panchasara reviewed (changes requested):
Overall, I like this approach of allowing the extension to control which option gets rendered. The UX would now have to explicitly state 'Open Trace File' or 'Open Trace Folder'
- In `trace-viewer-commands.ts`, could you rename  the label of the OpenTraceCommand to 'Open Trace Folder'? This is so that if a user decides to open a trace through the command palette they know what type of traces they can access. You could even add a command for 'Open Trace File' in this case.

ngondalia commented:
> Overall, I like this approach of allowing the extension to control which option gets rendered. The UX would now have to explicitly state 'Open Trace File' or 'Open Trace Folder'
> 
> * In `trace-viewer-commands.ts`, could you rename  the label of the OpenTraceCommand to 'Open Trace Folder'? This is so that if a user decides to open a trace through the command palette they know what type of traces they can access. You could even add a command for 'Open Trace File' in this case.

Thanks for reviewing! I have addressed your comments.

bhufmann commented on theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer-contribution.ts:
```diff
@@ -71,15 +73,16 @@ export class TraceViewerContribution
+        const selectFiles = type && type === TraceResourceType.FILE ? true : false;
         if (healthResponse && healthResponse.isOk() && healthResponse.getModel()?.status === 'UP') {
-            this.openDialog(rootPath);
+            this.openDialog(rootPath, selectFiles);
```
I think we should change the title of the dialog. Right now the title is `Open Trace` for both cases. It might not be clear for the user. Maybe change the title to `Open Trace File`/`Open Trace Folder` or `Select Trace File to Open`/`Select Trace Folder to Open`.

bhufmann commented on packages/react-components/src/trace-explorer/trace-explorer-placeholder-widget.tsx:
```diff
@@ -16,19 +22,26 @@ export class ReactExplorerPlaceholderWidget extends React.Component<ReactPlaceho
+            <div className="placeholder-open-workspace-button-container">
+                <button
+                    className="plcaeholder-open-workspace-button"
+                    title="Select a trace to open"
```
This should be different for each case to be clear to the user, maybe 'Select a single trace file to open' for file traces and 'Select a trace folder to open`

ngondalia replied:
Ok - I will update it in the next patch!

bhufmann commented:
@hriday-panchasara  could you please update your review?

hriday-panchasara reviewed (approved):
Looks good, thanks for the contribution!
